app/services/checksum_validator.rb

```diff
@@ -92,141 +87,202 @@ def moab_validator
                                           caller_validates_checksums: true)
   end
 
-  def validate_signature_catalog_listing
-    latest_signature_catalog_entries.each { |entry| validate_signature_catalog_entry(entry) }
-  rescue Errno::ENOENT
-    results.add_result(AuditResults::SIGNATURE_CATALOG_NOT_IN_MOAB, signature_catalog_path: latest_signature_catalog_path)
-  rescue Nokogiri::XML::SyntaxError
-    results.add_result(AuditResults::INVALID_MANIFEST, manifest_file_path: latest_signature_catalog_path)
-  end
+  def persist!(clear_connections: false)
+    # This is to deal with db connection timeouts.
+    ActiveRecord::Base.clear_active_connections! if clear_connections
 
-  def flag_unexpected_data_files
-    data_files.each { |file| validate_against_signature_catalog(file) }
+    transaction_ok = ActiveRecordUtils.with_transaction_and_rescue(results) do
+      yield if block_given?
+      complete_moab.save!
+    end
+    results.remove_db_updated_results unless transaction_ok
+    results.report_results
   end
 
-  # Adds to the AuditResults object for any errors in checksum validation it encounters.
-  # @param [Moab::StorageObjectVersion] moab_version
-  def validate_manifest_inventory(moab_version)
-    manifest_file_path = "#{moab_version.version_pathname}/#{MANIFESTS}/#{MANIFESTS_XML}"
-    parse_verification_subentities(moab_version.verify_manifest_inventory)
-  rescue Nokogiri::XML::SyntaxError
-    results.add_result(AuditResults::INVALID_MANIFEST, manifest_file_path: manifest_file_path)
-  rescue Errno::ENOENT
-    results.add_result(AuditResults::MANIFEST_NOT_IN_MOAB, manifest_file_path: manifest_file_path)
-  end
+  # Validates files on disk against the manifest inventory
+  class ManifestInventoryValidator
```

Comment:
I think this class is a great idea - but I would like it to be in a separate file with its own tests. Perhaps that is "phase 2" of this specific refactoring and will be in a separate PR?

Reply:
again, I don't know off the top of my head if ManifestInventoryValidator might be used elsewhere.   i suspect we test to see if there is a manifestInventory.xml file in some places, but not sure if that should live in this service.